app/services/calculators/state_benefits_calculator.rb

```diff
@@ -10,10 +10,17 @@ def call(state_benefits)
       def total_monthly_state_benefits(state_benefits)
         total = 0.0
         state_benefits.each do |state_benefit|
-          state_benefit.calculate_monthly_amount!
-          total += state_benefit.monthly_value unless state_benefit.exclude_from_gross_income
+          monthly_value = Calculators::MonthlyEquivalentCalculator.call(
+            assessment_errors: state_benefit.gross_income_summary.assessment.assessment_errors,
+            collection: state_benefit.state_benefit_payments,
+          )
+
+          # TODO: Stop persisting this
+          state_benefit.update!(monthly_value:)
+
+          total += monthly_value unless state_benefit.exclude_from_gross_income
         end
-        total
+        total.round(2)
```

Comment:
I'm unsure about this change - where should this rounding be done (if at all?)

Reply:
Ah, so the issue was some floating point rounding errors, and the root cause was some `.to_f` in the MonthlyIncomeConverter. Refactoring was causing some slightly _different_ floating behaviours, and the rounding was my clumsy attempt to fix them. The proper fix, which I've now done, is to stop doing `.to_f` on integers and replace them with `.to_d`